src/libs/pool-data-provider/hooks/use-incentives-data-context.tsx

```diff
@@ -0,0 +1,292 @@
+import { ComputedReserveData, IncentivesControllerInterface, TxBuilderV2 } from '@aave/protocol-js';
+import BigNumber from 'bignumber.js';
+import { ethers } from 'ethers';
+import React, { ReactNode, useContext, useEffect, useState } from 'react';
+import { useLocation } from 'react-router';
+import Preloader from '../../../components/basic/Preloader';
+import ErrorPage from '../../../components/ErrorPage';
+import { getProvider } from '../../../helpers/markets/markets-data';
+import { useApolloConfigContext } from '../../apollo-config';
+import {
+  PoolIncentivesWithCache,
+  useCachedIncentivesData,
+} from '../../caching-server-data-provider/hooks/use-cached-incentives-data';
+import { ConnectionMode, useConnectionStatusContext } from '../../connection-status-provider';
+import { useProtocolDataContext } from '../../protocol-data-provider';
+import { useDynamicPoolDataContext } from '../providers/dynamic-pool-data-provider';
+import { useStaticPoolDataContext } from '../providers/static-pool-data-provider';
+import { useCurrentTimestamp } from './use-current-timestamp';
+import {
+  IncentiveDataResponse,
+  ReserveIncentiveData,
+  useIncentivesData,
+  UserReserveIncentiveData,
+} from './use-incentives-data';
+
+// User incentives response type
+interface UserIncentiveDict {
+  [incentiveControllerAddress: string]: UserIncentiveData;
+}
+
+export interface UserIncentiveData {
+  rewardTokenAddress: string;
+  rewardTokenDecimals: number;
+  claimableRewards: BigNumber;
+  assets: string[];
+}
+
+export interface IncentivesDataContextData {
+  reserveIncentives: CalculateReserveIncentivesResponse[];
+  userIncentives: UserIncentiveDict;
+  txBuilder: IncentivesControllerInterface;
+}
+
+const IncentivesDataContext = React.createContext({} as IncentivesDataContextData);
+
+function formatBNInput(input: string, decimals: number): string {
+  let inputBN = new BigNumber(input);
+  inputBN = inputBN.shiftedBy(decimals);
+  return inputBN.toString();
+}
+
+// Calculate incentive token price from reserves data or priceFeed from UiIncentiveDataProvider
+function calculateRewardTokenPrice(
+  reserves: ComputedReserveData[],
+  address: string,
+  priceFeed: string,
+  priceFeedDecimals: number
+): string {
+  // For stkAave incentives, use Aave reserve data
+  if (address === '0x4da27a545c0c5b758a6ba100e3a049001de870f5') {
+    address = '0x7fc66500c84a76ad7e9c93437bfc5ac33e2ddae9';
+  }
+  // For WMATIC/WAVAX incentives, use MATIC/AVAX reserve data
+  if (
+    address === '0x0d500b1d8e8ef31e21c99d1db9a6444d3adf1270' ||
+    address === '0xb31f66aa3c1e785363f0875a1b74e27b85fd66c7'
+  ) {
+    address = '0xeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee';
+  }
+  const rewardReserve = reserves.find((reserve) => reserve.underlyingAsset === address);
+  if (rewardReserve) {
+    return formatBNInput(rewardReserve.price.priceInEth, Number(rewardReserve.decimals));
+  } else {
+    return formatBNInput(priceFeed, priceFeedDecimals);
```

Comment:
wouldn't it make sense to return the feedPrice if it exists instead of first trying it via a derivative token?